pinot-core/src/main/java/com/linkedin/pinot/core/data/GenericRow.java

```diff
@@ -53,18 +51,24 @@ public Object getValue(String fieldName) {
     return _fieldMap.get(fieldName);
   }
 
+  public void addField(String key, Object value) {
```

Comment:
Please check if this method is safe to use. Besides init(), GenericRow should be immutable. this addField() break this rule, and modify the map passed in in init() method.